pallets/subtensor/src/staking.rs

```diff
@@ -560,6 +560,13 @@ impl<T: Config> Pallet<T> {
         if !Self::hotkey_account_exists(hotkey) {
             Stake::<T>::insert(hotkey, coldkey, 0);
             Owner::<T>::insert(hotkey, coldkey);
+
+            // Update Owned map
+            let mut hotkeys = Owned::<T>::get(coldkey);
+            if !hotkeys.contains(hotkey) {
+                hotkeys.push(hotkey.clone());
+                Owned::<T>::insert(coldkey, hotkeys);
+            }
```

Comment:
This could be more efficient if `Owned` was a Map -> Set instead of Map -> Vec, maybe it makes more sense?

Reply:
Do you mean something like this ?

```rust
// Define OwnedHotkeys as a map to BTreeSet
pub type OwnedHotkeys<T: Config> = StorageMap<_, Blake2_128Concat, T::AccountId, BTreeSet<T::AccountId>, ValueQuery>;

// When updating:
OwnedHotkeys::<T>::mutate(coldkey, |hotkeys| {
    hotkeys.insert(hotkey.clone());
});
```

Since we dont care about order , we could even use a HashSet?

Reply:
Yeah exactly BTreeSet. Can't use a HashSet in the runtime because order is not deterministic I think, so not consensus safe.